### PR TITLE
High resolution timer

### PR DESCRIPTION
**Pull request by czapiga**
In this PR I will work on #327 issue. mips_clock_us will be incremented every mircosecond but for now I just wanted to check if changing types won't break anything.

### CODE REVIEW

**Comment by czapiga** [comment]
@cahirwpz Is this how I should actually change timer resolution? It seems like after going down to 1us tests are not even executed. I have to debug why kernel freezes but first I'd like to be sure I'm not doing something stupid with mips32_set_c0

**Comment by czapiga** [comment]
I noticed that I can go down with clock to ~20us on OVPsim and ~40us on qemu. Setting it to lower value makes kernel freeze in few different places

**Comment by czapiga** [comment]
It was called because of  `if (ms != _ms)` in `sys/clock.c`

File: mips/clock.c [hunk 1]
```diff
@@ -7,20 +7,19 @@
 #include <sched.h>
 #include <sysinit.h>
 
-/* This counter is incremented every millisecond. */
-static volatile realtime_t mips_clock_ms;
+/* System clock gets incremented every milisecond. */
+static timeval_t *sys_clock = &TIMEVAL_INIT(0, 0);
```
Comment (czapiga) [inline mips/clock.c]: How is this working? You take address of temporary object

**Comment by cahirwpz** [comment]
Well... we don't want the timer to cause [interrupt storm](https://en.wikipedia.org/wiki/Interrupt_storm)!!! I wanted `clock_get` to return exact time (based on `C0_COUNT`) without making any impact on the scheduler and callouts.

File: mips/clock.c (hunk 1 above)
Comment (cahirwpz) [inline mips/clock.c]: This is a part of C99 standard and is called *designated initializers*. It's not a *temporary*, but an *anonymous* value. On the top level it gets allocated in `.data` section, inside function it gets allocated on the stack.

**Comment by czapiga** [comment]
Ok, I think I get it now..

**Comment by rafalcieslak** [comment]
@czapiga Side note: ATM you are the only one in the team who uses the `not ready` label to mark unfinished PRs, others use `WIP` for that. cahirwpz, me and coodie have been using  `not ready` to mark tasks and PRs that require work in other areas, extra discussion, or additional research before work on such task can even begin.

Based on what we typically use labels for, I've created [this table](https://github.com/cahirwpz/mimiker/wiki/Github-issue-PR-labels) some time ago - it may serve as a nice overview of what do we usually mean with each label. I'm not forcing you to strictly follow these descriptions, though. It is not an officially accepted list. We may want to discuss it in future to reduce confusion and hopefully improve our labeling consistency.

**Comment by cahirwpz** [comment]
@czapiga What is the progress? I'd like this task to be finished sooner than later. It will unblock #253 and enable us to add `gettimeofday` syscall.

**Comment by czapiga** [comment]
I will try to end this till Thursday.

**Comment by czapiga** [comment]
If it's possible I would like this to be merged then I will finish #253 and after that I will go back to `timer_event_t` interface.

**Pull request merged**